feature-libs/checkout/base/core/facade/checkout-payment.service.spec.ts

```diff
@@ -0,0 +1,338 @@
+      expect(resultState).toEqual({
+        loading: false,
+        error: false,
+        data: mockCardTypes,
+      });
+      subscription.unsubscribe();
+    }));
+  });
+
+  describe(`getCardTypes`, () => {
+    it(`should call facade's getCardTypesState()`, (done) => {
+      spyOn(service, 'getCardTypesState').and.returnValue(
+        of({
+          loading: false,
+          error: false,
+          data: mockCardTypes,
+        })
+      );
+
+      service
+        .getCardTypes()
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual(mockCardTypes);
+          done();
+        });
+    });
+
+    it(`should return an empty array if query's data is falsy`, (done) => {
+      spyOn(service, 'getCardTypesState').and.returnValue(
+        of({
+          loading: false,
+          error: false,
+          data: undefined,
+        })
+      );
+
+      service
+        .getCardTypes()
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual([]);
+          done();
+        });
+    });
+  });
+
+  describe(`getPaymentDetailsState`, () => {
+    it(`should return the delivery modes`, (done) => {
+      spyOn(checkoutQuery, 'getCheckoutDetailsState').and.returnValue(
+        of(<QueryState<CheckoutState>>{
+          loading: false,
+          error: false,
+          data: {
+            paymentInfo: mockPaymentInfo,
+          },
+        })
+      );
+
+      service
+        .getPaymentDetailsState()
+        .pipe(take(1))
+        .subscribe((result) => {
+          expect(result).toEqual(<QueryState<PaymentDetails | undefined>>{
+            loading: false,
+            error: false,
+            data: mockPaymentInfo,
+          });
+          done();
+        });
+    });
+  });
+
+  describe(`createPaymentDetails`, () => {
+    it(`should call checkoutPaymentConnector.create`, () => {
+      spyOn(connector, 'create').and.stub();
+
+      service.createPaymentDetails(mockPaymentInfo);
+
+      expect(connector.create).toHaveBeenCalledWith(
+        mockUserId,
+        mockCartId,
+        mockPaymentInfo
+      );
+    });
+
+    it(`should dispatch PaymentDetailsCreatedEvent event`, () => {
+      spyOn(eventService, 'dispatch').and.stub();
+
+      service.createPaymentDetails(mockPaymentInfo);
+
+      expect(eventService.dispatch).toHaveBeenCalledWith(
+        {
+          userId: mockUserId,
+          cartId: mockCartId,
+          paymentDetails: mockPaymentInfo,
+        },
+        PaymentDetailsCreatedEvent
+      );
+    });
+
+    // TODO: Replace with event testing once we remove ngrx store.
+    it(`should dispatch UserActions.LoadUserPaymentMethods`, () => {
+      spyOn(store, 'dispatch').and.stub();
+
+      service.createPaymentDetails(mockPaymentInfo);
+
+      expect(store.dispatch).toHaveBeenCalledWith(
+        new UserActions.LoadUserPaymentMethods(mockUserId)
+      );
+    });
+  });
+
+  describe(`setPaymentDetails`, () => {
```

Comment:
add test case for should throw error when paymentDetailsId is not provided

Reply:
added